html/interactive-guides/microprofile-config/download-deployment-exception.html

```diff
@@ -0,0 +1,18 @@
+<!DOCTYPE html>
+<!--
+  Copyright (c) 2017 IBM Corporation and others.
+  All rights reserved. This program and the accompanying materials
+  are made available under the terms of the Eclipse Public License v1.0
+  which accompanies this distribution, and is available at
+  http://www.eclipse.org/legal/epl-v10.html
+ 
+  Contributors:
+      IBM Corporation - initial API and implementation
+-->
+ <html>
+
+<head>
+</head>
+<body>
+    <p style="color: red;">DeploymentException:  the default value has NOT been specified for the <code>download_url</code> configuration property</p>
```

Comment:
I just tried it with a real case of not putting a defaultValue for a ConfigProperty and the property is not set in any config. Here's the error:

[ERROR   ] CWWKZ0002E: An exception occurred while starting the application io.openliberty.guides.microprofile.mpconfig. The exception message was: com.ibm.ws.container.service.state.StateChangeException: org.jboss.weld.exceptions.DeploymentException: WELD-001408: Unsatisfied dependencies for type String with qualifiers @ConfigProperty
  at injection point [BackedAnnotatedField] @Inject @ConfigProperty private download_url

I don't think the implementation can actually catch this error as an exception and remap it. Hmmm, this is a start application error. It is not an error to be displayed in web browser. We're going down the wrong path to display this error. We need a terminal to have the interaction to start the server and get the error. Here's the full console display from starting the server:

[lavena@Lavenas-MacBook-Pro:bin]$ ./server run defaultServer
Java HotSpot(TM) 64-Bit Server VM warning: ignoring option MaxPermSize=256m; support was removed in 8.0
objc[73600]: Class JavaLaunchHelper is implemented in both /Library/Java/JavaVirtualMachines/jdk1.8.0_144.jdk/Contents/Home/jre/bin/java (0x10bbc14c0) and /Library/Java/JavaVirtualMachines/jdk1.8.0_144.jdk/Contents/Home/jre/lib/libinstrument.dylib (0x10bcbb4e0). One of the two will be used. Which one is undefined.
Launching defaultServer (Open Liberty 17.0.0.3/wlp-1.0.18.cl170320170927-1854) on Java HotSpot(TM) 64-Bit Server VM, version 1.8.0_144-b01 (en_US)
[AUDIT   ] CWWKE0001I: The server defaultServer has been launched.
[AUDIT   ] CWWKZ0058I: Monitoring dropins for applications. 
[WARNING ] CWNEN0057W: The io.openliberty.guides.microprofile.InventoryConfig.portNumber injection target must not be declared static.
[ERROR   ] CWMCG5003E: The [BackedAnnotatedField] @Inject @ConfigProperty private io.openliberty.guides.microprofile.InventoryConfig.testString InjectionPoint dependency was not resolved. Error: java.util.NoSuchElementException: CWMCG0015E: The property io.openliberty.guides.microprofile.test was not found in the configuration.
	at com.ibm.ws.microprofile.config.impl.AbstractConfig.getValue(AbstractConfig.java:129)
	at [internal classes]

[ERROR   ] CWMCG5003E: The [BackedAnnotatedField] @Inject @ConfigProperty private io.openliberty.guides.microprofile.InventoryConfig.testString InjectionPoint dependency was not resolved. Error: java.util.NoSuchElementException: CWMCG0015E: The property io.openliberty.guides.microprofile.test was not found in the configuration.
	at com.ibm.ws.microprofile.config.impl.AbstractConfig.getValue(AbstractConfig.java:129)
	at [internal classes]

[ERROR   ] CWWKZ0002E: An exception occurred while starting the application io.openliberty.guides.microprofile.mpconfig. The exception message was: com.ibm.ws.container.service.state.StateChangeException: org.jboss.weld.exceptions.DeploymentException: WELD-001408: Unsatisfied dependencies for type String with qualifiers @ConfigProperty
  at injection point [BackedAnnotatedField] @Inject @ConfigProperty private io.openliberty.guides.microprofile.InventoryConfig.testString
  at io.openliberty.guides.microprofile.InventoryConfig.testString(InventoryConfig.java:0)

[AUDIT   ] CWWKF0012I: The server installed the following features: [servlet-3.1, jndi-1.0, json-1.0, cdi-1.2, jsonp-1.0, mpConfig-1.1, jaxrs-2.0, jaxrsClient-2.0].
[AUDIT   ] CWWKF0011I: The server defaultServer is ready to run a smarter planet.